package/MDAnalysis/analysis/rms.py

```diff
@@ -535,29 +554,79 @@ def __init__(self, atomgroup, reference=None, select='all',
-                             "weights=None or weights='mass', not a weight "
-                             "array.")
+        # check weights type
+
+        if self.weights is None:
+            pass
+        elif (type(self.weights) == str) or \
+            (np.array(self.weights).ndim == 1) and (np.array(self.weights).dtype 
+                                                in (np.dtype('float64'),np.dtype('int64'))):
+            self.__weight_type_check(self.weights, self.mobile_atoms, select)
+        else:       # list of weights conditions
+            if len(self.weights) != len(self.groupselections) + 1:   #length check
+                raise ValueError("Length of array of weights is not equal to " 
+                                       "length of groupselections + 1")
+            weights_list = self.weights
+            if weights_list[0] is None:
+                pass
+            elif type(weights_list[0]) == str or \
+                ((np.array(weights_list[0]).ndim == 1) and (np.array(weights_list[0]).dtype 
+                                                in (np.dtype('float64'),np.dtype('int64')))):
+                self.__weight_type_check(weights_list[0], self.mobile_atoms, select)
+            for weights, atoms, selection in zip(weights_list[1:],
+                        self._groupselections_atoms,self.groupselections):
+                self.__weight_type_check(weights, atoms['mobile'], selection)      
 
     def _prepare(self):
         self._n_atoms = self.mobile_atoms.n_atoms
-
-        if not iterable(self.weights) and self.weights == 'mass':
-            self.weights = self.ref_atoms.masses
-        if self.weights is not None:
-            self.weights = np.asarray(self.weights, dtype=np.float64) / np.mean(self.weights)
+        if str(self.weights) == 'mass':         # apply 'mass' weights for all selections
+           self.weights = ['mass'] * (len(self.groupselections) + 1)
+        elif self.weights is None:         # apply 'None' weights for all selections
+           self.weights = [None] * (len(self.groupselections) + 1)  
+        elif (np.array(self.weights).ndim == 1) and (np.array(self.weights).dtype 
+                                             in (np.dtype('float64'),np.dtype('int64'))):
+           none_selection = [None] * len(self.groupselections)       # apply '1D array' weights to select
+           none_selection.insert(0, self.weights)                   # & apply 'None' weights to groupselections
+           self.weights = none_selection
+
+
+
+        # add the array of weights to weights_select
+        if str(self.weights[0]) == 'mass':
+            self.weights_select = self.mobile_atoms.masses
+            self.weights_ref = self.ref_atoms.masses
+        elif self.weights[0] is None:
+            self.weights_select = None
+            self.weights_ref = None
+        else:
+            self.weights_select = self.weights[0]
+            self.weights_ref = self.weights[0]
```

Comment:
You could use `get_weights` with `self.weights[0]` and each `self.mobile_atoms` and `self.ref_atoms` to achieve this more simply